iac-extensions/arm/src/test/java/org/sonar/iac/arm/visitors/ArmSymbolVisitorTest.java

```diff
@@ -0,0 +1,459 @@
+  private static final Map<String, String> VARIABLE_DECLARATION = Map.of(
+    BICEP, "var foo = 'bar'",
+    JSON, "\"foo\": \"bar\"");
+  private static final Map<String, String> VARIABLE_DECLARATION_WITH_USAGE = Map.of(
+    BICEP, "var bar = '${foo}'",
+    JSON, "\"bar \": \"[variables('foo')]\"");
+
+  public static Set<String> languagesToTest() {
+    return Set.of(BICEP, JSON);
+  }
+
+  @ParameterizedTest
+  @MethodSource("languagesToTest")
+  void registeredTreesShouldBeVisited(String language) {
+    String code = fileWithDefaultVariableDeclaration(language);
+
+    File file = parse(code);
+    List<String> visited = new ArrayList<>();
+    ArmSymbolVisitor visitor = new ArmSymbolVisitor();
+
+    visitor.register(File.class, (ctx, tree) -> visited.add("file_visit"));
+    visitor.register(VariableDeclaration.class, (ctx, tree) -> visited.add("variable_declaration_visit"));
+    visitor.registerAfter(Identifier.class, (ctx, tree) -> visited.add("identifier_visit_after"));
+    visitor.registerAfter(File.class, (ctx, tree) -> visited.add("file_visit_after"));
+    visitor.scan(inputFileContext, file);
+
+    assertThat(visited).containsExactly(
+      "file_visit",
+      "variable_declaration_visit",
+      "file_visit_after",
+      "identifier_visit_after");
+  }
+
+  @ParameterizedTest
+  @MethodSource("languagesToTest")
+  void variableDeclarationShouldCreateSymbol(String language) {
+    String code = fileWithDefaultVariableDeclaration(language);
+
+    File file = scanFile(code);
+    HasSymbol declaration = (HasSymbol) file.statements().get(0);
+    SymbolTable symbolTable = file.symbolTable();
+
+    assertThat(symbolTable).isNotNull();
+    assertThat(symbolTable.hasFoundUnresolvableVariableAccess()).isFalse();
+    assertThat(symbolTable.getSymbols()).hasSize(1);
+    assertThat(symbolTable.getSymbol("bar")).isNull();
+
+    Symbol symbol = symbolTable.getSymbol("foo");
+    assertThat(symbol).isNotNull().isEqualTo(declaration.symbol());
+    assertThat(symbol.name()).isEqualTo("foo");
+    assertThat(symbol.symbolTable()).isEqualTo(symbolTable);
+
+    assertThat(symbol.usages()).allSatisfy(usage -> {
+      assertThat(usage.kind()).isEqualTo(Usage.Kind.ASSIGNMENT);
+      assertThat(usage.tree()).isEqualTo(declaration);
+    });
+  }
+
+  @ParameterizedTest
+  @CsvSource(delimiterString = ";", value = {
+    // BICEP
+    // access in variableDeclaration should create usage
+    "bicep;VAR;" + "var bar =  '${foo}'",
+    "bicep;VAR;" + "var bar =  '${foo}ConcatToVariable'",
+    // access in function in variableDeclaration should create usage
+    "bicep;VAR;" + "var bar =  '${toLower(foo)}ConcatToVariable'",
+
+    // output
+    "bicep;OUT;" + "output foo string =  foo",
+
+    "json;VAR;" + "\"bar\": \"[variables('foo')]\"",
+    "json;VAR;" + "\"bar\": \"[concat(variables('foo'), '-addToVar')]\"",
+    "json;VAR;" + "\"bar\": \"[concat(toLower(variables('foo')), '-addToVar')]\""
```

Comment:
I guess the string concatenation here is used for better formatting the code and parameters visibility?
Maybe it is better to use `@MethodSource` and JUnit5 Arguments? Then the parameters are separated naturally by language syntax

Reply:
Yes I agree. I already started this in my next PR for Parameters, but I will change it here as well!